src/server/scripts/Commands/cs_debug.cpp

```diff
@@ -1593,6 +1604,216 @@ class debug_commandscript : public CommandScript
+            return true;
+        }
+
+        return false;
+    }
+
+    static bool HandleHoverCheatCommand(ChatHandler* handler, const char* args)
+    {
+        Unit* unit = handler->getSelectedUnit();
+        if (unit == nullptr)
+            unit = handler->GetSession()->GetPlayer();
+
+        if (unit == nullptr)
+            return false;
+
+        std::string argstr = (char*)args;
+
+        if (!*args)
+            argstr = unit->IsHovering() ? "off" : "on";
+
+        if (argstr == "off")
+        {
+            unit->SetHover(false);
+            handler->SendSysMessage("Hover is OFF on the selected unit (or self if no target).");
+            return true;
+        }
+        else if (argstr == "on")
+        {
+            unit->SetHover(true);
+            handler->SendSysMessage("Hover is ON on the selected unit (or self if no target).");
+            return true;
+        }
+
+        return false;
+    }
+
+    static bool HandleCanTransitionBetweenSwimAndFlyCheatCommand(ChatHandler* handler, const char* args)
+    {
+        Unit* unit = handler->getSelectedUnit();
+        if (unit == nullptr)
+            unit = handler->GetSession()->GetPlayer();
+
+        if (unit == nullptr)
+            return false;
+
+        std::string argstr = (char*)args;
+
+        if (!*args)
+            argstr = unit->CanTransitionBetweenSwimAndFly() ? "off" : "on";
+
+        if (argstr == "off")
+        {
+            unit->SetCanTransitionBetweenSwimAndFly(false);
+            handler->SendSysMessage("CanTransitionBetweenSwimAndFly is OFF on the selected unit (or self if no target).");
+            return true;
+        }
+        else if (argstr == "on")
+        {
+            unit->SetCanTransitionBetweenSwimAndFly(true);
+            handler->SendSysMessage("CanTransitionBetweenSwimAndFly is ON on the selected unit (or self if no target).");
+            return true;
+        }
+
+        return false;
+    }
+
+    static bool HandleCanFlyCheatCommand(ChatHandler* handler, const char* args)
+    {
+        Unit* unit = handler->getSelectedUnit();
+        if (unit == nullptr)
+            unit = handler->GetSession()->GetPlayer();
+
+        if (unit == nullptr)
+            return false;
+
+        std::string argstr = (char*)args;
+
+        if (!*args)
+            argstr = unit->HasCanFly() ? "off" : "on";
+
+        if (argstr == "off")
+        {
+            unit->SetCanFly(false);
+            handler->SendSysMessage("CanFly is OFF on the selected unit (or self if no target).");
+            return true;
+        }
+        else if (argstr == "on")
+        {
+            unit->SetCanFly(true);
+            handler->SendSysMessage("CanFly is ON on the selected unit (or self if no target).");
+            return true;
+        }
+
+        return false;
+    }
+
+    static bool HandleFeatherFallCheatCommand(ChatHandler* handler, const char* args)
+    {
+        Unit* unit = handler->getSelectedUnit();
```

Comment:
there should be a getSelectedUnitOrSelf() function that already returns the current player if no selected Unit is found. If there isn't you can add it, similar to getSelectedPlayerOrSelf()